com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/util/ApplicationURITest.java

```diff
@@ -24,9 +34,100 @@
     @MethodSource("testParameters")
     public void testGetHostDomainPath(String uri, String expectedHost, String expectedDomain, String expectedPath) {
         ApplicationURI applicationURI = new ApplicationURI(uri);
-        Assertions.assertEquals(expectedDomain, applicationURI.getDomain());
-        Assertions.assertEquals(expectedDomain, ApplicationURI.getDomainFromURI(uri));
-        Assertions.assertEquals(expectedHost, applicationURI.getHost());
+        assertParametersAreValid(expectedDomain, applicationURI.getDomain(), expectedDomain, ApplicationURI.getDomainFromURI(uri),
+                                 expectedHost, applicationURI.getHost());
+        Assertions.assertEquals(expectedPath, applicationURI.getPath());
+    }
+
+    private static Stream<Arguments> routeTestParameters() {
+        return Stream.of(Arguments.of(createCloudRoute("custom-host", createCloudDomain("custom-domain"), null), "custom-host",
+                                      "custom-domain", null),
+                         Arguments.of(createCloudRoute("custom-host", createCloudDomain("custom-domain"), "/custom-path"), "custom-host",
+                                      "custom-domain", "/custom-path"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("routeTestParameters")
+    public void testGetHostDomainPath(CloudRoute route, String expectedHost, String expectedDomain, String expectedPath) {
+        ApplicationURI applicationURI = new ApplicationURI(route);
+        assertParametersAreValid(expectedDomain, applicationURI.getDomain(), expectedDomain, route.getDomain()
+                                                                                                  .getName(),
+                                 expectedHost, applicationURI.getHost());
         Assertions.assertEquals(expectedPath, applicationURI.getPath());
     }
+
+    @Test
+    public void testGetURIParts() {
+        ApplicationURI applicationURI = new ApplicationURI(createCloudRoute("custom-host", createCloudDomain("custom-domain"), null));
+        Map<String, Object> expectedParts = new HashMap<>();
+        expectedParts.put(SupportedParameters.HOST, "custom-host");
+        expectedParts.put(SupportedParameters.DOMAIN, "custom-domain");
+        expectedParts.put(SupportedParameters.ROUTE_PATH, null);
```

Comment:
Why do we need to insert a null value when when the checking for equality be done, the missing parameter will anyways return null?

Reply:
Because getURIParts will put host, domain, and path regardless of their values. So if route_path is not put the maps won't be equal.